File: java/Inheritance.java
class Weapon {
  protected String Type = "Sword";
  public void Sound() { 
    System.out.println("Shing, shing!");
  }
}

class Inheritance extends Weapon {
  private String modelName = "excalibur";
  public static void main(String[] args) {

    Inheritance mySword = new Inheritance();

    mySword.Sound();

    System.out.println(mySword.Type + " " + mySword.modelName);
  }
}

File: dart_application_1/bin/inheritance.dart
class Weapon {
  String type = "Sword";
  void sound() {
    print("Shing, shing");
  }
}

class Sword extends Weapon {
  String Name = "Excalibur";
} 

void main() {
  Sword mySword = Sword();
  mySword.sound();
  print(mySword.type + " " + mySword.Name);
}